### DISCUSSION TITLE
Notification warning for muted sound increases volume to 100 instead of unmuting it

### DISCUSSION THREAD

**Discussion by ArchUsr64** [ppy/osu#15980] [Q&A] [answered]
The notification warning when trying to play while music is muted, increases both master and music volume to 100 instead of unmuting the sound when clicked. There also seems to be a bug(or intended design?) where this notification warning warns the user only once per session.

https://user-images.githubusercontent.com/83179501/145055657-a08f36ff-40ad-45c8-9017-4e1a1e536195.mp4

**Comment by bdach** [comment, marked as answer]
both things were intentional at implementation time:

* for the "restore to 100% volume" part, see commit message of https://github.com/ppy/osu/commit/78931c8e23f304e0b39387b6d8d355ff64c01555
* the notification is only shown once during a session to avoid nagging the user too much.

**Reply by ArchUsr64** [reply]
I think the behaviour should be different based on whether the sound is muted(like in the video clip) or if just the master or music volume is set to 0.

**Reply by bdach** [reply]
maybe the master/music/effect volumes should only reset to default if they're 0? not sure...

**Reply by peppy** [reply]
Can agree with it only unmuting in such a case.

The goal of the notification is that a lot of people start the game from desktop and don't realise their volume is zero, then report bugs. The intention is to fix this common scenario, and as above, not nag the user each time this happens (they may intentionally have the game muted for whatever reason).

**Reply by peppy** [reply]
Will be addressed by #15999.
